File: HackerRank/1- 30 Days of Code/Day 26 Nested Logic/answer.py
# Enter your code here. Read input from STDIN. Print output to STDOUT
return_day, return_month, return_year = [
    int(e) for e in input().strip().split(' ')]
due_day, due_month, due_year = [int(e) for e in input().strip().split(' ')]

if due_year > return_year:
    print(0)

elif return_year == due_year:

    # check month
    if due_month > return_month:
        print(0)

    elif due_month == return_month:

        # check day
        if due_day >= return_day:
            print(0)
        else:
            print(15 * (return_day - due_day))

    else:
        print(500 * (return_month - due_month))
else:
    print(10000)

File: HackerRank/1- 30 Days of Code/Day 26 Nested Logic/answer.js
function processData(input) {
  //splitting variables
  let inputArray = input.split("\n");
  let returnedDate = inputArray[0].split(" ");
  let dueDate = inputArray[1].split(" ");

  // store days, months, and years
  let day = 0,
    month = 1,
    year = 2,
    fine = 0;

  // check if late or not
  if (returnedDate[year] > dueDate[year]) {
    fine = 10000;
  } else if (returnedDate[month] > dueDate[month]) {
    fine = 500 * (returnedDate[month] - dueDate[month]);
  } else if (returnedDate[day] > dueDate[day]) {
    fine = 15 * (returnedDate[day] - dueDate[day]);
  }

  console.log(fine);
}

process.stdin.resume();
process.stdin.setEncoding("ascii");
_input = "";
process.stdin.on("data", function (input) {
  _input += input;
});

process.stdin.on("end", function () {
  processData(_input);
});
